domain/fhir/eoverdracht/builders.go

```diff
@@ -242,6 +251,11 @@ func (b FHIRBuilder) BuildNursingHandoffComposition(patient *Patient, advanceNot
 	return b.buildNursingHandoffComposition(administrativeData, careplan, fhirPatient), nil
 }
 
-func (FHIRBuilder) generateResourceID() string {
-	return uuid.New().String()
+type IDGenerator interface {
+	String() string
 }
+
+type UUIDGenerator struct {}
+func (UUIDGenerator) String() string {
+	return uuid.NewString()
+}
```

Comment:
I'm not sure why we would want or need an interface here but apart from that it doesn't make sense to call it a `Generator` as it only contains a `String()` method which is not generating anything but simply converting a value to a string.

Reply:
Looking at it I would even remove the `TransferFHIRBuilder` interface and `FHIRBuilder` structure and convert all `Build*` methods to simple functions.

Reply:
Change the Generator.String() -> to something morre specific.